core/src/main/java/io/lionweb/lioncore/java/serialization/JsonSerialization.java

```diff
@@ -496,19 +479,16 @@ private List<ClassifierInstance<?>> deserializeClassifierInstances(
             new MapBasedResolver(deserializedByID), this.instanceResolver);
     serializedClassifierInstances.stream()
         .forEach(
-            n -> {
+            node -> {
               populateClassifierInstance(
-                  n, serializedToInstanceMap.get(n), classifierInstanceResolver);
-              ClassifierInstance<?> classifierInstance = serializedToInstanceMap.get(n);
+                  node, serializedToInstanceMap.get(node), classifierInstanceResolver);
+              ClassifierInstance<?> classifierInstance = serializedToInstanceMap.get(node);
               if (classifierInstance instanceof AnnotationInstance) {
-                SerializedAnnotationInstance serializedAnnotationInstance =
-                    (SerializedAnnotationInstance) n;
-                if (serializedAnnotationInstance == null) {
+                if (node == null) {
                   throw new IllegalStateException(
                       "Dangling annotation instance found (annotated node is null). ");
                 }
-                Node annotatedNode =
-                    (Node) deserializedByID.get(serializedAnnotationInstance.getParentNodeID());
+                Node annotatedNode = (Node) deserializedByID.get(node.getParentNodeID());
```

Comment:
Can an annotations have children? If so the parent of an annotation could be another annotation and this could fail

Reply:
I don't know if annotations can have children. (This change doesn't impact that, though.) I'll also check the TS-impl. for what we do there.

Reply:
An `Annotation` is a `Classifier` so can own `Feature`s. It can't own another `Annotation`, only reference them through its `extends` feature. So, I'd say the code as it is, is OK. (In the TS-impl. it wouldn't be a problem anyway, because of the way that parent-child-relations are installed.)